server/superdesk/aap_mm_datalayer.py

```
@@ -115,15 +138,29 @@ def find_one_raw(self, resource, _id):
             del doc['fetch_endpoint']
 
         # Only if we have credentials can we download the original if the account has that privilege
-        if 'AAP_MM_USER' in self._app.config and 'AAP_MM_PASSWORD' in self._app.config \
-                and self._app.config['AAP_MM_USER'] is not None:
-            url = self._app.config['AAP_MM_SEARCH_URL'] + '/Assets/{}/Original/download'.format(_id)
+        if self._username is not None and self._password is not None:
+            if doc['type'] == 'picture':
+                url = self._app.config['AAP_MM_SEARCH_URL'] + '/Assets/{}/Original/download'.format(_id)
+                mime_type = 'image/jpeg'
+            elif doc['type'] == 'video':
+                resolutions = self._get_resolutions(_id)
+                if any(v['Name'] == 'Ipod' for v in resolutions['Video']):
+                    url = self._app.config['AAP_MM_SEARCH_URL'] + '/Assets/{}/Ipod/download'.format(_id)
+                    mime_type = doc.get('renditions').get('original').get('mimetype')
+                else:
+                    raise NotImplementedError
```

We should raise a specific error in this case I think.